tests/base/src/test/java/org/wildfly/security/http/client/ElytronHttpClientCertTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.http.client;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.wildfly.security.auth.client.ElytronXmlParser;
+import org.wildfly.security.auth.client.InvalidAuthenticationConfigurationException;
+import org.wildfly.security.http.client.utils.ClientCertSSLTestUtils;
+import org.wildfly.security.auth.client.AuthenticationContext;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.http.HttpResponse;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivilegedAction;
+import java.security.cert.CertificateException;
+
+import static java.security.AccessController.doPrivileged;
+
+/**
+ * Test for the ElytronHttpClient class
+ *
+ * @author <a href="mailto:kekumar@redhat.com">Keshav Kumar</a>
+ */
+public class ElytronHttpClientCertTest {
+    static final String RESOURCES = "./target/keystores/";
+    private static SSLServerSocketTestInstance sslServerSocketTestInstancePort10001;
+
+    @BeforeClass
+    public static void before() throws NoSuchAlgorithmException, KeyStoreException, IOException, CertificateException {
+        ClientCertSSLTestUtils.createKeystores();
+
+        sslServerSocketTestInstancePort10001 = new SSLServerSocketTestInstance(RESOURCES + "server1.keystore.jks", RESOURCES + "server1.truststore.jks", 10001);
+
+        sslServerSocketTestInstancePort10001.run();
+    }
+
+    @AfterClass
+    public static void after() {
+        sslServerSocketTestInstancePort10001.stop();
+
+        ClientCertSSLTestUtils.deleteKeystores();
+    }
+
+    @Test
+    public void testElytronHttpClientCertAuth(){
+        getAuthenticationContext("wildfly-config-http-client-cert.xml").run(() -> {
+           try {
+               String uri = "https://localhost:10001";
+               ElytronHttpClient.ElytronHttpHeaderBuilder builder = new ElytronHttpClient.ElytronHttpHeaderBuilder(uri);
+               HttpResponse httpResponse = builder.connect();
```

Comment:
@keshav-725 Just some thought on this. Previously, I was thinking of this:

```
ElytronHttpClientBuilder clientBuilder = new ElytronHttpClientBuilder();

ElytronHttpClient client = clientBuilder.url(myUrl)
            .addHeader(myHeader)
            .build();
client.connect();
```

But seeing this written out it is not a good idea to have request details configured for the client instance. It is better if the request details are detached from the client configuration. So the client can make requests to different endpoints without reconfiguration.

So I propose this usage:

```
ElytronHttpClient client = new ElytronHttpClient();
HttpRequest httpRequest = HttpRequestBuilder.url(myUrl)
            .addHeader(myHeader)
            .addMehtod(POST)
            .build();
client.connect(httpRequest);
```

But since HttpRequestBuilder already exists in java https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpRequest.Builder.html we do not have to introduce a new builder but use the existing one and so allow for the connect method to accept HttpRequest parameter as I mentioned on the proposal before:

```
public HttpResponse connect(HttpRequest httpRequest) {
...
}
```

It can then be used like this:

```
ElytronHttpClient client = new ElytronHttpClient();
HttpRequest request = HttpRequest.newBuilder()
  .uri(new URI("myURI"))
  .POST()
  .header(myKey,myValue)
  .build();

client.connect(request)
```

Or we can later introduce our new wrapper for the HttpRequest, like ElytronHttpRequest that can be used with clients that do not use Java's HttpRequest if this will be needed. But this can be introduced later if needed along with the changes that should be added to make the ElytronHttpClient more generic and possible to use with other clients than just java's HttpClient.

Let me know if this makes sense? Thanks!